tests/js/client/shell/shell-restore-integration.js

```diff
@@ -1292,126 +1154,105 @@ function restoreIntegrationSuite() {
       });
 
       const path = fs.getTempFile();
-      try {
-        fs.makeDirectory(path);
+      fs.makeDirectory(path);
 
-        const dbName = 'UnitTestRestoreRegressionDb';
-        db._createDatabase(dbName);
-        createDumpJsonFile(path, dbName);
+      const dbName = 'UnitTestRestoreRegressionDb';
+      db._createDatabase(dbName);
+      createDumpJsonFile(path, dbName);
 
-        for (const colJson of collectionsJson) {
-          const colName = colJson.parameters.name;
-          const fn = fs.join(path, colName + ".structure.json");
-          fs.write(fn, JSON.stringify(colJson));
-        }
+      for (const colJson of collectionsJson) {
+        const colName = colJson.parameters.name;
+        const fn = fs.join(path, colName + ".structure.json");
+        fs.write(fn, JSON.stringify(colJson));
+      }
 
-        const args = ['--server.database', dbName, '--import-data', 'false'];
-        runRestore(path, args, 0);
+      const args = ['--server.database', dbName, '--import-data', 'false'];
+      runRestore(path, args, 0);
 
-        db._useDatabase(dbName);
-        for (const colJson of collectionsJson) {
-          const col = db._collection(colJson.parameters.name);
-          assertInstanceOf(arangodb.ArangoCollection, col);
-          assertEqual(colJson.parameters.name, col.name());
-          assertEqual(colJson.parameters.type, col.type());
-          if (isCluster) {
-            assertEqual(colJson.parameters.distributeShardsLike, col.properties().distributeShardsLike);
-          }
+      db._useDatabase(dbName);
+      for (const colJson of collectionsJson) {
+        const col = db._collection(colJson.parameters.name);
+        assertInstanceOf(arangodb.ArangoCollection, col);
+        assertEqual(colJson.parameters.name, col.name());
+        assertEqual(colJson.parameters.type, col.type());
+        if (isCluster) {
+          assertEqual(colJson.parameters.distributeShardsLike, col.properties().distributeShardsLike);
         }
-        fs.removeDirectoryRecursive(path, true);
-      } finally {
-        db._useDatabase("_system");
       }
```

Comment:
There are a few places where you left the `_useDatabase` in a finally. Was this intentional? Should we keep the finally here, or can we remove the try-finally in the other cases as well?

Reply:
I guess not being in the system database is potentially bad for following up tests, hence if the test changes the database, it should change it back?
but probably `teardown` could be doing that as well, but I don't think we should do this in the backport.

Reply:
Agreed about not doing this in the backport. I just wanted to point out that the changes are not fully consistent.